operator/controllers/loki/internal/lokistack/ruler_config_discovery.go

```diff
@@ -19,23 +19,48 @@ const (
 // to the named Lokistack in the same namespace of the RulerConfig. If no LokiStack is found, then
 // skip reconciliation.
 func AnnotateForRulerConfig(ctx context.Context, k k8s.Client, name, namespace string) error {
-	var s lokiv1.LokiStack
 	key := client.ObjectKey{Name: name, Namespace: namespace}
-
-	if err := k.Get(ctx, key, &s); err != nil {
-		if apierrors.IsNotFound(err) {
-			// Do nothing
-			return nil
-		}
-
-		return kverrors.Wrap(err, "failed to get lokistack", "key", key)
+	ss, err := getLokiStack(ctx, k, key)
+	if err != nil {
+		return err
 	}
```

Comment:
AFAIU if the returned lokistack is nil (i.e. because the api returns `IsNotFound`) the following code will still try to update the annotation. Shouldn't we add an early return when `ss == nil` to omit update to err?